native/cocos/audio/oalsoft/AudioEngine-soft.cpp

```diff
@@ -616,7 +616,9 @@ ccstd::vector<uint8_t> AudioEngineImpl::getOriginalPCMBuffer(const char *url, ui
             tmpBuf = static_cast<char *>(malloc(audioInfo.bytesPerFrame * framesToReadOnce));
             do {
                 framesRead = decoder->read(framesToReadOnce, tmpBuf); //read one by one to easy divide
-                pcmData.reserve(totalFrames + framesRead);
+                pcmData.resize(bytesPerChannelInFrame * (audioInfo.totalFrames+ framesRead));
+                p = pcmData.data();
+                p += bytesPerChannelInFrame * audioInfo.totalFrames;
```

Comment:
Do we need this hacking code for windows platform? The original issue ( https://github.com/cocos2d/cocos2d-x/issues/16938 ) happened only on iOS / macOS.